src/terminal/terminal.c

```diff
@@ -1240,7 +1253,8 @@ static void __guac_terminal_resize(guac_terminal* term, int width, int height) {
             /* Update buffer top and cursor row based on shift */
             term->buffer->top += shift_amount;
             term->cursor_row  -= shift_amount;
-            term->visible_cursor_row  -= shift_amount;
+            if (term->visible_cursor_row != -1)
```

Comment:
Any reason a sentinel value of -1 is being used vs. the `cursor_visible` flag?

If -1 is going to be a value assigned to `visible_cursor_row` and `visible_cursor_col` to indicate that the cursor is not rendered, that should be documented where those properties are defined, however I'm not sure of the utility of a -1 constant when there is a boolean flag also in use and with clear semantics.

If the -1 sentinel is not needed after all, and `cursor_visible` is preferred, it would be good to document that the values of the `visible_cursor_row`, etc. properties have no meaning if `cursor_visible` is `true`.

Reply:
While most of the time `cursor_visible == false` is equivalent to `visible_cursor_... == -1`, it is not always the case. In particular when we just switched visibility back, we can have `cursor_visible = true` and `visible_cursor_...` still at -1. In that case this condition would change `visibile_curor_row` from `-1` to `-1 - shift_amount` which would be semantically incorrect.

Similarly, on line 788 if we used `cursor_visible` we could trigger a buffer underflow by trying to access row/column -1.

Documenting those properties makes sense.

An alternative would be to introduce an additional `bool visible_cursor_visible` that would track the effective visibility of the cursor (while `cursor_visible` tracks the requested visibility), the proposed naming here is consistent with `visible_cursor_...` tracking the effective state of `cursor_...`.